xbmc/games/addons/GameClientMouse.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ *      Copyright (C) 2015-2016 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "GameClientMouse.h"
+#include "GameClient.h"
+#include "GameClientTranslator.h"
+#include "addons/kodi-addon-dev-kit/include/kodi/kodi_game_types.h"
+#include "input/InputManager.h"
+#include "input/Key.h"
+#include "utils/log.h"
+
+using namespace GAME;
+
+#define BUTTON_INDEX_MASK  0x01ff
+
+CGameClientMouse::CGameClientMouse(const CGameClient* gameClient, const GameClient* dllStruct) :
+  m_gameClient(gameClient),
+  m_dllStruct(dllStruct)
+{
+  CInputManager::GetInstance().RegisterMouseHandler(this);
+}
+
+CGameClientMouse::~CGameClientMouse()
+{
+  CInputManager::GetInstance().UnregisterMouseHandler(this);
+}
+
+bool CGameClientMouse::OnMouseMotion(int x, int y)
+{
+  // Only allow activated input in fullscreen game
+  if (!m_gameClient->AcceptsInput())
+  {
+    return false;
+  }
+
+  bool bHandled = false;
+
+  game_input_event event;
+
+  event.type            = GAME_INPUT_EVENT_RELATIVE_POINTER;
+  event.port            = -1;
+  event.controller_id   = ""; // TODO
+  event.feature_name    = ""; // TODO
+  event.rel_pointer.x   = x;
+  event.rel_pointer.y   = y;
+
+  try
+  {
+    bHandled = m_dllStruct->InputEvent(&event);
+  }
+  catch (...)
+  {
+    CLog::Log(LOGERROR, "GAME: %s: exception caught in InputEvent()", m_gameClient->ID().c_str());
+  }
+
+  return bHandled;
+}
+
+bool CGameClientMouse::OnMouseButtonPress(MOUSE::BUTTON button)
+{
+  // Only allow activated input in fullscreen game
+  if (!m_gameClient->AcceptsInput())
+  {
+    return false;
+  }
+
+  bool bHandled = false;
+
+  game_input_event event;
+
+  event.type                   = GAME_INPUT_EVENT_DIGITAL_BUTTON;
+  event.port                   = -1;
```

Comment:
Here i'm not sure if that's the correct event type / port. Because the code in https://github.com/kodi-game/game.libretro/pull/7/files#diff-97dc480240fc0f2c40a1732397d7adecR148 now feels like a hack now. (checking for -1)